src/code/z_player_lib.c

```diff
@@ -1,153 +1,1021 @@
+        player->modelAnimType = 0;
+    }
+
+    Player_SetModels(player, modelGroup);
+}
+
+void func_80123C58(Player* player) {
+    player->heldItemActionParam = player->itemActionParam;
+    Player_SetModelGroup(player, Player_ActionToModelGroup(player, player->itemActionParam));
+    player->unk_AA5 = 0;
+}
+
+void Player_SetEquipmentData(GlobalContext* globalCtx, Player* player) {
+    if (player->csMode != 0x86) {
+        player->currentShield = CUR_EQUIP_VALUE_VOID(EQUIP_SHIELD);
+        if ((player->transformation != PLAYER_FORM_ZORA) || (((player->currentBoots != PLAYER_BOOTS_ZORA_LAND)) &&
+                                                             (player->currentBoots != PLAYER_BOOTS_ZORA_UNDERWATER))) {
+            player->currentBoots = D_801BFF90[player->transformation];
+        }
+        Player_SetModelGroup(player, Player_ActionToModelGroup(player, player->itemActionParam));
+        func_80123140(globalCtx, player);
+        if (player->unk_B62 != 0) {
+            player->unk_B62 = 1;
+        }
+    }
+}
+
+void func_80123D50(GlobalContext* globalCtx, Player* player, s32 item, s32 actionParam) {
+    func_80114FD0(globalCtx, item, player->heldItemButton);
+
+    if (item != ITEM_BOTTLE) {
+        player->heldItemId = item;
+        player->itemActionParam = actionParam;
+    }
+
+    player->heldItemActionParam = actionParam;
+}
+
+void func_80123DA4(Player* player) {
+    player->unk_730 = NULL;
+    player->stateFlags2 &= ~PLAYER_STATE2_2000;
+}
+
+void func_80123DC0(Player* player) {
+    if ((player->actor.bgCheckFlags & 1) ||
+        (player->stateFlags1 & (PLAYER_STATE1_8000000 | PLAYER_STATE1_800000 | PLAYER_STATE1_200000)) ||
+        (!(player->stateFlags1 & (PLAYER_STATE1_80000 | PLAYER_STATE1_40000)) &&
+         ((player->actor.world.pos.y - player->actor.floorHeight) < 100.0f))) {
+        player->stateFlags1 &= ~(PLAYER_STATE1_40000000 | PLAYER_STATE1_80000 | PLAYER_STATE1_40000 |
+                                 PLAYER_STATE1_20000 | PLAYER_STATE1_10000 | PLAYER_STATE1_8000);
+    } else if (!(player->stateFlags1 & (PLAYER_STATE1_200000 | PLAYER_STATE1_80000 | PLAYER_STATE1_40000))) {
+        player->stateFlags1 |= PLAYER_STATE1_80000;
+    } else if ((player->stateFlags1 & PLAYER_STATE1_40000) && (player->transformation == PLAYER_FORM_DEKU)) {
+        player->stateFlags1 &=
+            ~(PLAYER_STATE1_40000000 | PLAYER_STATE1_20000 | PLAYER_STATE1_10000 | PLAYER_STATE1_8000);
+    }
+    func_80123DA4(player);
+}
+
+void func_80123E90(GlobalContext* globalCtx, Actor* actor) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    func_80123DC0(player);
+    player->unk_730 = actor;
+    player->unk_A78 = actor;
+    player->stateFlags1 |= PLAYER_STATE1_10000;
+    func_800DFD78(Play_GetCamera(globalCtx, CAM_ID_MAIN), 8, actor);
+    Camera_ChangeMode(Play_GetCamera(globalCtx, CAM_ID_MAIN), 9);
+}
+
+s32 func_80123F14(GlobalContext* globalCtx) {
```

Comment:
Do any of these functions you changed to take globalCtx not match with state? Is that why they're changed?

Reply:
iirc, yes